ICSharpCode.Decompiler/IL/ControlFlow/SwitchAnalysis.cs

```diff
@@ -230,15 +231,36 @@ bool MatchSwitchVar(ILInstruction inst)
 				return inst.MatchLdLoc(out switchVar);
 		}
 
+		bool MatchSwitchVar(ILInstruction inst, out long sub)
+		{
+			if (inst.MatchBinaryNumericInstruction(BinaryNumericOperator.Sub, out var left, out var right) && right.MatchLdcI(out sub))
```

Comment:
I'm assuming Roslyn uses a plain `sub` here? You should also check the overflow flag, as-is you're also matching `sub.ovf`/`sub.ovf.un`.